model/Signal.py

```diff
@@ -39,19 +39,57 @@ def get_features_window(self, x, list_features="all"):
             features = np.concatenate((features, f(x).reshape(1, 8)), axis=1)
         return features
 
-    def remove_transition(self):
-        s = self.get_features(list_features=["mav"]).sum(axis=1)
-        # detection
-        algo = rpt.Binseg(model="l2").fit(s)
-        result = algo.predict(n_bkps=1)[0]
-        if result > len(s)//2:
-            result = 20
-        trans_idx = result*self.step
-        if self.signal.shape[0]>2*trans_idx:
-            self.signal = self.signal[trans_idx:, :]
-            self.n_samples = self.signal.shape[0]
+    def remove_transition(self, algorithm="Binseg", w=50, min_s=30):
+        
+        if algorithm == "Binseg":
+
+            # detection using Binseg
+            s = self.get_features(list_features=["mav"]).sum(axis=1)
+            algo = rpt.Binseg(model="l2").fit(s)
+            result = algo.predict(n_bkps=1)[0]
+            if result > len(s) // 2:
+                result = 20
+            trans_idx = result * self.step
+            if self.signal.shape[0] > 2 * trans_idx:
+                self.signal = self.signal[trans_idx:, :]
+                self.n_samples = self.signal.shape[0]
+
+        if algorithm == "Pelt":
+
+            # store original signal
+            original_signal = self.signal.copy()
+
+            # extract features using sliding average
+            self.signal = self.sliding_avg(self.signal, w)
+            s = self.get_features(list_features=["mav"]).sum(axis=1)
+
+            # initialize change point detection with Pelt
+            result = None
 
+            # attempt valid removal, otherwise make window shorter to make the algorithm less aggressive
+            while min_s > 5 and not result:
+                try:
+                    detect = rpt.Pelt(model="l2", min_size=min_s).fit(s)
+                    result = detect.predict(pen=20)
+                    if result:  # valid removal found
+                        break
+                except Exception:  # if window size too large 
+                    min_s -= 1  # decrease min_size 
 
+            # find first big change in direction
+            if result:
+                cutting_index = result[0]
+            else:
+                # cut the signal at 10% of its length if no cutting point is found with minimal window
+                cutting_index = int(len(original_signal) * 0.1)
```

Comment:
try to follow the syntax in the original function when you're transforming the signal. For example the index is called trans_idx in the original. Because your function is doing the same thing (removing transition) as the original, it should look like the old one in it's logic and variable names so a third person reading it can quickly make sense of what's going on.